src/extra/handlebars/HandlebarsEngine.ts

```diff
@@ -150,11 +167,60 @@ export class HandlebarsEngine
     );
   }
 
-  compile(template: string) {
+  enableProfileDataLayer(): void {
+
+    this.engine.registerHelper("profileData", function (opts: ProfileDataHelperOptions) {
+
+      // See https://blog.osteele.com/2007/12/cheap-monads/
+      const $N = {};
+      // Check if we have the value in the DataLayer
+      if ((((((opts || $N).data || $N).root || $N).private || $N).profileData || $N)[opts.hash.fieldName]) {
+        return opts.data.root.private.profileData[opts.hash.fieldName];
```

Comment:
That looks awfully close to a vendor specific implementation.

Reply:
Do you want me to remove it?

At first, it look terrible, but when writing it/reading it after the facts, it's not such a nightmare.